lib/ContractInteract/GatewayComposer.js

```diff
@@ -0,0 +1,445 @@
+   * @param {string} brandedToken Address of brandedToken contract.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @returns {Promise<GatewayComposer>} Promise containing the gateway composer
+   *                                     instance that has been deployed.
+   */
+  static async deploy(
+    web3,
+    owner,
+    valueToken,
+    brandedToken,
+    txOptions,
+  ) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address: ${txOptions.from}.`);
+      return Promise.reject(err);
+    }
+
+    const tx = GatewayComposer.deployRawTx(
+      web3,
+      owner,
+      valueToken,
+      brandedToken,
+    );
+
+    return Utils.sendTransaction(tx, txOptions).then((txReceipt) => {
+      const address = txReceipt.contractAddress;
+      return new GatewayComposer(web3, address);
+    });
+  }
+
+  /**
+   * Raw transaction for {@link GatewayComposer#deploy}.
+   *
+   * @param {Web3} web3 Origin chain web3 object.
+   * @param {string} owner Address of the staker on the value chain.
+   * @param {string} valueToken Address of value token contract.
+   * @param {string} brandedToken Address of brandedToken contract.
+   *
+   * @returns {Object} Raw transaction.
+   */
+  static deployRawTx(
+    web3,
+    owner,
+    valueToken,
+    brandedToken,
+  ) {
+    if (!(web3 instanceof Web3)) {
+      throw new TypeError(
+        `Mandatory Parameter 'web3' is missing or invalid: ${web3}`,
+      );
+    }
+    if (!Web3.utils.isAddress(valueToken)) {
+      throw new TypeError(`Invalid valueToken address: ${valueToken}.`);
+    }
+    if (!Web3.utils.isAddress(brandedToken)) {
+      throw new TypeError(`Invalid branded token address: ${brandedToken}.`);
+    }
+
+    const abiBinProvider = new AbiBinProvider();
+    const bin = abiBinProvider.getBIN(ContractName);
+
+    const args = [
+      owner,
+      valueToken,
+      brandedToken,
+    ];
+
+    const contract = Contracts.getGatewayComposer(web3);
+
+    return contract.deploy(
+      {
+        data: bin,
+        arguments: args,
+      },
+    );
+  }
+
+  /**
+   * Method for request stake. This transfers value tokens from caller to
+   * GatewayComposer.
+   *
+   * @param {string} stakeVT Stake amount in wei.
+   * @param {string} mintBT Minted amount in wei.
+   * @param {string }gateway Address of gateway contract.
+   * @param {string} beneficiary Beneficiary address on auxiliary chain.
+   * @param {string} gasPrice Gas price that staker is willing to pay for the reward.
+   * @param {string} gasLimit Maximum gas limit for reward calculation.
+   * @param {string} nonce Staker nonce managed by gateway contract.
+   * @param txOptions Transaction options.
+   *
+   * @returns {Promise<Object>} Promise that resolves to transaction receipt.
+   */
```

Comment:
We should mention in documentation as dev note that "requestStake is called by staker after GC is approved for ValueToken."